utils/web_utils.go

```diff
@@ -168,3 +170,13 @@ func GetJwks(ctx context.Context, location string) (jwk.Set, error) {
 		return key, nil
 	}
 }
+
+// Equivalent to c.ClientIP() except that it returns netip.Addr instead of net.IP
+//
+// If the IP string is invalid, it returns netip.Addr{}. Use netip.Addr.ISValid() to
+// check the validity of the returned value
+func ClientIPAddr(c *gin.Context) netip.Addr {
```

Comment:
Can you leave a comment here to mention that this also checks the `X-Real-IP`? I see it does from the tests and the docs, but it's nice to know that at a glance.